test/cppsuite/test_harness/workload/workload_validation.h

```diff
@@ -75,66 +81,80 @@ class workload_validation {
          * checked in check_reference.
          */
         for (auto const &it : deleted_collections) {
-            if (!verify_collection_state(session, it, false))
+            if (!verify_collection_file_state(session, it, false))
                 testutil_die(DEBUG_ERROR,
-                  "validate: collection %s present on disk while it has been tracked as deleted.",
+                  "Validation failed: collection %s present on disk while it has been tracked as "
+                  "deleted.",
                   database::build_collection_name(it).c_str());
         }
 
+        /*
+         * All collections in memory should match those created in the schema tracking table.
+         * Dropping is currently not supported.
+         */
+        std::sort(created_collections.begin(), created_collections.end());
+        auto on_disk_collection = created_collections.begin();
+        for (const auto &it : known_collection_ids) {
```

Comment:
Let's call them `id` instead of `it` since they're not iterators which is what I think led to @etienneptl's confusion below.

Also, there's no point using `&` since that is to avoid copy constructing each element. However, it is a container of integers it's the same as copying pointers.

Reply:
815301699